### DISCUSSION TITLE
No post run capability for composite actions?

### DISCUSSION THREAD

**Discussion by pcolmer** [community/community#26743] [Actions]
<p>According to <a href="https://docs.github.com/en/free-pro-team@latest/actions/creating-actions/metadata-syntax-for-github-actions#runs-for-composite-run-steps-actions" rel="noopener nofollow ugc">Metadata syntax for GitHub Actions - GitHub Docs</a>, it looks like there isn’t the ability to specify steps for the action to be executed “post run”. This capability exists for Javascript and Docker actions but not composite.</p>
<p>Is that correct? Just making sure I’ve correctly interpreted the documentation.</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/pcolmer">@pcolmer</a>,</p>
<blockquote>
<p>Is that correct? Just making sure I’ve correctly interpreted the documentation.</p>
</blockquote>
<p>Yes, it is correct. The <strong>pre</strong> and <strong>post</strong> option are not available to the composite run steps actions.</p>
<p>Generally, you can think of the composite action as just a combination of some <strong>run</strong> steps. When you run the a  composite action in the workflow, it is functionally equivalent to directly executing the combined <strong>run</strong> steps in the workflow.</p>
<p>If you want to run some <strong>pre</strong> or <strong>post</strong> scripts for the composite action, you can directly combine the run as a step in the action:</p>
<ul>
<li>
<p>Set the first step in the composite action to <strong>run</strong> some <strong>pre</strong> scripts to set up some preparations for the subsequent <strong>run</strong> steps in the composite action.</p>
</li>
<li>
<p>Set the last step in the composite action to <strong>run</strong> some <strong>post</strong> scripts to cleanup some data generated by the previous <strong>run</strong> steps in the composite action.</p>
</li>
</ul>

**Comment by pcolmer** [comment]
<p>Thanks for the clarification/confirmation.</p>
<aside class="quote group-github-partners" data-username="brightran" data-post="2" data-topic="139046">
<div class="title">
<div class="quote-controls"></div>
 brightran:</div>
<blockquote>
<p>Generally, you can think of the composite action as just a combination of some <strong>run</strong> steps. When you run the a composite action in the workflow, it is functionally equivalent to directly executing the combined <strong>run</strong> steps in the workflow.</p>
</blockquote>
</aside>
<p>Unfortunately (for me), I was primarily wanting to use the composite action as a means of creating an action that allowed a broader support of languages other than Docker and Javascript. So, in that regard, I needed the <strong>post</strong> script to always run in the workflow even if a previous step had failed.</p>
<p>Ultimately, I ended up going with a Javascript Action.</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/pcolmer">@pcolmer</a>,</p>
<blockquote>
<p>I needed the  <strong>post</strong>  script to always run in the workflow even if a previous step had failed.</p>
</blockquote>
<p>Unlike directly executing the <strong>run</strong> steps in the job, the option “<strong>runs.steps.if</strong>” is not supported. So you can’t set an <strong>if</strong> conditional (e.g. <strong>if: always()</strong>) to let a run step to always run when any previous <strong>run</strong> steps is failed in the composite action.</p>

**Comment by keroro520** [comment]
<p>Hope <code>post</code> for composite action.</p>

**Comment by stewartie4** [comment]
<p>Post scripts from actions that your composite action executes still run<br>
so even though post is not supported directly you can use something like this</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://github.com/webiny/action-post-run">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">

      <a href="https://github.com/webiny/action-post-run" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/18306778/181099703-38163cd8-52e2-4ab7-a5df-5f696cfde2a5.png" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/4/d/4ddf9b1dd6201748858c95e9d4e3aaac3052552f_2_690x345.png, https://aws1.discourse-cdn.com/github/optimized/3X/4/d/4ddf9b1dd6201748858c95e9d4e3aaac3052552f_2_1035x517.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/4/d/4ddf9b1dd6201748858c95e9d4e3aaac3052552f.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/4/d/4ddf9b1dd6201748858c95e9d4e3aaac3052552f_2_10x10.png"></div>

<h3><a href="https://github.com/webiny/action-post-run" target="_blank" rel="noopener nofollow ugc">GitHub - webiny/action-post-run: Enables executing custom commands once a...</a></h3>

  <p>Enables executing custom commands once a workflow job has ended. - GitHub - webiny/action-post-run: Enables executing custom commands once a workflow job has ended.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
<p>
to get it to work</p>

**Comment by jetexe** [comment]
As another option, you can use [run-and-post-run-action](https://github.com/marketplace/actions/run-and-post-run-action), which allows you to define both `run` and `post` commands, and also supports multiple command execution.
